container/go/pkg/compat/ociToLegacy.go

```diff
@@ -80,15 +83,35 @@ func LegacyFromOCIImage(img v1.Image, srcDir, dstDir string) error {
 		layerDigest, err := layer.Digest()
 		if err != nil {
 			return errors.Wrap(err, "failed to fetch the layer's digest")
-	}
+		}
 
 		layerPath = path.Join(targetDir, layerDigest.Hex)
-		out := strconv.Itoa(i) + targzExt
+		out := strconv.Itoa(i) + compressedLayerExt
 		dstLink = path.Join(dstDir, out)
 		if err = generateSymlinks(layerPath, dstLink); err != nil {
 			return errors.Wrapf(err, "failed to generate legacy symlink for layer %d with digest %s", i, layerDigest)
 		}
 	}
 
+	// symlink for the image manifest pulled into OCI layout to .json, so must rename the file from being
+	// named after its sha256 digest under blobs/sha256 to manifest.json
+	imgIndex, err := layout.ImageIndexFromPath(srcDir)
+	if err != nil {
+		return errors.Wrapf(err, "unable to open image as index from %s", srcDir)
+	}
+	manifest, err := imgIndex.IndexManifest()
+	if err != nil {
+		return errors.Wrap(err, "unable to get manifest from image index")
+	}
+	if len(manifest.Manifests) != 1 {
+		log.Fatal("error: image contains more than one manifest")
```

Comment:
Update message `log.Fatalf("Image index read from %s had unexpected number of manifests: got %d, want 1.", srcDir, len(manifest.Manifests))`